spec/integration/cds_importer/entity_mapper_spec.rb

```diff
@@ -90,5 +92,84 @@
       mapper.import
       expect(AdditionalCode.last.filename).to eq 'test.gzip'
     end
+
+    context 'when measureExcludedGeographicalArea changes are present' do
+      let(:xml_node) do
+        {
+          'sid' => '20130650',
+          'validityStartDate' => '2021-01-01T00:00:00',
+          'metainfo' => {
+            'opType' => 'C',
+            'origin' => 'T',
+            'status' => 'L',
+            'transactionDate' => '2021-02-01T17:42:46',
+          },
+          'measureExcludedGeographicalArea' =>
+          [
+            {
+              'metainfo' => {
+                'opType' => 'C',
+                'origin' => 'T',
+                'status' => 'L',
+                'transactionDate' => '2021-02-01T17:42:46',
+              },
+              'geographicalArea' => {
+                'hjid' => '23808',
+                'sid' => '439',
+                'geographicalAreaId' => 'CN',
+                'validityStartDate' => '1984-01-01T00:00:00',
+              },
+            },
+          ],
+        }
+      end
+
+      let(:mapper) { described_class.new('Measure', xml_node) }
+
+      let(:measure) {
+        create(:measure, measure_sid: '20130650')
+      }
+
+      it 'does not remove excluded geographical areas that belong to measures not present within the XML increment' do
+        create(:geographical_area, geographical_area_sid: '439')
+        other_exclusion = create(:measure_excluded_geographical_area)
+
+        mapper.import
+
+        expect(MeasureExcludedGeographicalArea[measure_sid: other_exclusion.measure_sid]).to be_present
+      end
+
+      context 'when there is an existing exclusion for this measure' do # rubocop:disable RSpec/NestedGroups
```

Comment:
Should we chuck this in the config file? It would be significant amount of rearchitecting to correct tis nested groups violation and I can read your tests fine.

Reply:
Hmm, we could do that.